### ISSUE TITLE
Agenda - New Event from Peruser.php give me a SQL Error

### ISSUE THREAD

**Issue by maverjk** [Dolibarr/dolibarr#11771]
Hello

I'm testing the new Dolibarr version 10.0.1 and I found out an issue, maybe someone can help me to sort it, or maybe is a bug that there is the need to sort out. However creating a New Event in Agenda from Peruser.php form give me following error:

> SELECT id, code, type, libelle as label, color, active, picto FROM llx_c_actioncomm WHERE code='AC_ATT_REG'
> INSERT INTO llx_actioncomm(datec,datep,datep2,durationp,fk_action,code,fk_soc,fk_project,note,fk_contact,fk_user_author,fk_user_action,fk_user_done,label,percent,priority,fulldayevent,location,punctual,transparency,fk_element,elementtype,entity,extraparams,email_msgid,email_from,email_sender,email_to,email_tocc,email_tobcc,email_subject,errors_to) VALUES ('2019-08-27 17:42:52', '2019-08-26 00:00:00', null, null, 51,'AC_ATT_REG', null, null,  '', null, 1, 2, null, 'Presenza','-1','0','1','','1', '0', null, null, 1,null, null, null, null, null, null, null, null, null)
> INSERT INTO llx_actioncomm_resources(fk_actioncomm, element_type, fk_element, mandatory, transparency, answer_status) VALUES(37, 'user', 2, 0, 0, 0)
> INSERT INTO llx_actioncomm_resources(fk_actioncomm, element_type, fk_element, mandatory, transparency, answer_status) VALUES(37, 'user', , 0, 0, 0)
> [DB_ERROR_SYNTAX] You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ' 0, 0, 0)' at line 1 

I tried to generate a New Event from other Agenda forms and appears fine, instead here (Peruser.php) I got this problem. 

As suggested from the Error, SQL is missing of a parameter. Other "strange thing" to my eyes is the fact that from log I can actually see another INSERT INTO llx_actioncomm_resources that is well formed before the one that is giving error.

Any advice where I can find a kind of clue?

**Comment by latherton** [comment]
It appears the issue is that the "fk_element" field is null in the last insert query (VALUES(37, 'user', , 0, 0, 0)). This field in the database does not allow null values.

I have 2 questions....
A. Are you creating an event for a user other than yourself?
B. Are you adding yourself to the event?

If the answer to these questions are "yes" then add yourself to the event and see if that works.

**Comment by maverjk** [comment]
Hi 
To answer to your questions I placed others test.
Well (case 1) if I'm going to add an event for "logged in user" in peruser.php form, I got that current logged in user is automatically selected as event owner in card.php, and it actually works.
Instead (case 2) if I'm going to add an event for "another user" clicking on any little square in peruser.php form, I got that this user is assigned two times in event owner field in card.php (first time as owner, second time as user) and now I have error:
> You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near ' 0, 0, 0)' at line 1

Digging more from "case 2", let's call it "case 3", I saw also that if I'm going to delete the "another user" selected two times, and I'm going to add him/her again as well as any other user different from "logged in user", another error come out when I'm going to create the event:
>  ErrorPropertyUserowneridNotDefined

Finally starting form previous point (case 3) if I select current "logged in user", event got created correctly as in "case 1".
